### PR TITLE
Bug fix for Scatter3D lines+markers with line color array error

### LINKED ISSUE #2873: Scatter3D lines+markers with line color array error
For the `scatter3d` trace type, an error condition occurs when the `line.color` property is specified as a numeric array *and* `markers` is included in the `mode` flaglist.

Issue originally reported in of https://github.com/plotly/plotly.py/issues/1093

CodePen: https://codepen.io/anon/pen/ZjjORB

```javascript
x = [0, 0, 0]
y = [1, 1, 1]
z = [1, 2, 3]
c = [0, 1, 2]
var trace1 = {
    type: 'scatter3d',
    x: x,
    y: y,
    z: z,
    // mode: "lines",
    mode: "lines+markers",
    line: {
        color: c,
        showscale: true,
        width: 10
    }
}


var data = [ trace1 ];

var layout = {};

Plotly.newPlot('myDiv', data, layout);
```

When running example against `plotly-latest.js` I get the following JavaScript console error

```
TypeError: inputContainer is undefined

doUpdate
    https://cdn.plot.ly/plotly-latest.js:79899:13
calc
    https://cdn.plot.ly/plotly-latest.js:79940:5
calcMarkerColorscale
    https://cdn.plot.ly/plotly-latest.js:156566:13
calc
    https://cdn.plot.ly/plotly-latest.js:158689:5
calci
    https://cdn.plot.ly/plotly-latest.js:127889:22
[795]</plots.doCalcdata
    https://cdn.plot.ly/plotly-latest.js:127913:42
[719]</exports.plot
    https://cdn.plot.ly/plotly-latest.js:102574:16
[719]</exports.newPlot
    https://cdn.plot.ly/plotly-latest.js:102973:12
```

### PR DESCRIPTION
**Pull request by archmoj**
Fix #2873 
@etpinard

### CODE REVIEW

File: src/components/colorscale/calc.js [hunk 1]
```diff
@@ -33,7 +33,12 @@ module.exports = function calc(gd, trace, opts) {
             updateStyle(trace._input, containerStr ? (containerStr + '.' + attr) : attr, inputVal);
         }
         else {
-            inputContainer[attr] = inputVal;
+            if(trace._input.type === 'scatter3d') {
+                trace._input[attr] = inputVal;
```
Comment (etpinard) [inline src/components/colorscale/calc.js]: Hmm. Looks to me like `containerStr` about isn't getting set properly? What's its value here?

Comment (archmoj) [reply src/components/colorscale/calc.js]: The `inputContainer` was `undefined` if I recall correctly.

Comment (etpinard) [reply src/components/colorscale/calc.js]: Right , but what about `containerStr`?

From this block above

https://github.com/plotly/plotly.js/blob/3ea5c82de6790a3471fea10774fc50b9c3023ae6/src/components/colorscale/calc.js#L50-L54

this leads be to believe that `containerStr` isn't getting set properly.

Comment (archmoj) [reply src/components/colorscale/calc.js]: First it is equal to `line`; but then when it is set to `marker` we get an error.

Comment (etpinard) [reply src/components/colorscale/calc.js]: > but then when it is set to `marker` we get an error.

Right, and in this case `Colorscale.calc` should get called in the first place. Something must be not working correctly here:

https://github.com/plotly/plotly.js/blob/3ea5c82de6790a3471fea10774fc50b9c3023ae6/src/traces/scatter/colorscale_calc.js#L27-L34

**Comment by etpinard** [comment]
@archmoj PR https://github.com/plotly/plotly.js/pull/3341 fixes this bug. More specifically commit https://github.com/plotly/plotly.js/pull/3341/commits/c9d92d7942d6eae4eaf7e204e43c5dbf90441369 removes all the `inputContainer` logic in `Colorscale.calc` and commit https://github.com/plotly/plotly.js/pull/3341/commits/e8973faca5269c7bdc881a197b7a426164f91d58 adds a new mock :lock:ing down the fix.

Closing!

**Comment by archmoj** [comment]
Excellent!

**Pull request closed without merging**